src/infinitecanonicalmps.jl

```diff
@@ -48,11 +60,82 @@ function shift_flux(i::Index, flux_density::QN)
   return ITensors.setspace(i, shift_flux(space(i), flux_density))
 end
 
+function multiply_flux(qn::ITensors.QNVal, flux_factor::Int64)
+  if qn.modulus == 0 #is undefined, so remains undefined
+    return (qn.name, qn.val, qn.modulus) #qn
+  elseif qn.modulus == 1 || qn.modulus == -1 #U(1) symmetry, should remain a U(1) symmetry
+    return (qn.name, qn.val * flux_factor, qn.modulus) #qn * flux_factor
+  else #To get the same algebra, multiplying the modulus by flux_factor works
+    return (qn.name, qn.val * flux_factor, qn.modulus * flux_factor)
+  end
+end
+
+function multiply_flux(qn::QN, flux_factor::Int64)
+  flux_factor == 1 && return qn
+  return QN(multiply_flux.(filter(x -> x.modulus != 0, qn.data), flux_factor)...)
```

Comment:
Can't the logic of checking `flux_factor == 1` and `x.modulus != 0` be handled by `multiply_flux` (to be renamed `scale_flux`)?

Also, why would one of the QNs have a modulus of 0? It seems like that should be caught somewhere else and not filtered/checked here.